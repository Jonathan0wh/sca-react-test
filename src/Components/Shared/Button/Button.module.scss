@import "styles/colors";

.button {
  width: 80px;
  height: 36px;
  background-color: $grey;
  color: $white;
  padding-top: 10px;
  padding-bottom: 10px;
  padding-left: 10px;
  padding-right: 10px;
  border-radius: 4px;
  border-color: $grey;
  font-size: small;
  text-align: center;
  cursor: pointer;
}
