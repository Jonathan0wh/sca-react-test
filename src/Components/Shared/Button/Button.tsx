import React from 'react';
import PulseLoader from 'react-spinners/PulseLoader';

import styles from './Button.module.scss';

interface ButtonProps {
  buttonText: string;
  handleClick: Function;
  loading?: boolean;
}

const Button = ({ buttonText, handleClick, loading }: ButtonProps) => (
  <button
    className={styles.button}
    onClick={event => {
      event.preventDefault();
      handleClick();
    }}
  >
    {loading ? <PulseLoader size={6} color="white" /> : buttonText}
  </button>
);

export default Button;
