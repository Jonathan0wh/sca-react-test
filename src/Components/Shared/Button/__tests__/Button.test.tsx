import React from 'react';
import { shallow } from 'enzyme';

import Button from '../Button';

describe('<Button />', () => {
  const clickFn = jest.fn();

  it('render button in normal condtion', () => {
    const wrapper = shallow(<Button buttonText="test" handleClick={clickFn} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render button in loading condtion', () => {
    const wrapper = shallow(
      <Button buttonText="test" handleClick={clickFn} loading />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
