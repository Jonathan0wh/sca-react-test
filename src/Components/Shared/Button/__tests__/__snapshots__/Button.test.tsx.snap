// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render button in loading condtion 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  <Loader
    color="white"
    css={Object {}}
    loading={true}
    margin="2px"
    size={6}
    sizeUnit="px"
  />
</button>
`;

exports[`<Button /> render button in normal condtion 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  test
</button>
`;
